components/micro-gateway-core/src/main/ballerina/src/gateway/handlers/mutualSSL_handler.bal

```diff
@@ -58,98 +58,155 @@ public type MutualSSLHandler object {
+                return false;
+            }
+            return true;
+        } else {
+            handle|error certificateAlias = getAliasFromRequest(cert.toString());
+            if (certificateAlias is error) {
+                setErrorMessageToInvocationContext(API_AUTH_GENERAL_ERROR);
+                return prepareAuthenticationError("Unclassified Authentication Failure");
+            }
+            if (certificateAlias is handle) {
+                //Only if 'isValidateCertificateWithAPI' true then certificate should be cross checked with the
+                //alias list provided in the config. Otherwise no need to cross check with the alias list
+                //in micro-gw.conf.
+                boolean isExistAlias = (isValidateCertificateWithAPI) ? isExistApiAlias(apiVersion, apiName,
+                                                        certificateAlias.toString(),self.apiCertificateList) : true;
+                if (!isExistAlias || certificateAlias.toString() == "") {
+                    if(!isExistAlias) {
+                        printError(KEY_AUTHN_FILTER, "Mutual SSL authentication failure. API is not associated " +
+                    "with the certificate");
+                    } else if (certificateAlias.toString() == "") {
+                        printError(KEY_AUTHN_FILTER, "Mutual SSL authentication failure. Certificate alias not " +
+                        "found in the trust store");
                     }
-                    if (certificateAlias is handle ) {
-                        boolean isExistAlias = isExistApiAlias(apiVersion, apiName, certificateAlias.toString(),
-                        self.apiCertificateList);
-                        if (!isExistAlias || certificateAlias.toString() == "") {
-                            printDebug(KEY_AUTHN_FILTER, "Mutual SSL authentication failure. API is not associated " +
-                            "with the certificate");
-                            self.gatewayCache.addMutualSslCertificateCache(cacheKey, false);
-                            setErrorMessageToInvocationContext(API_AUTH_INVALID_CREDENTIALS);
-                            return false;
+                    self.gatewayCache.addMutualSslCertificateCache(cacheKey, false);
+                    setErrorMessageToInvocationContext(API_AUTH_INVALID_CREDENTIALS);
+                    return false;
+                }
+            }
+            self.gatewayCache.addMutualSslCertificateCache(cacheKey, true);
+            printDebug(KEY_AUTHN_FILTER, "MutualSSL handshake status using request context: PASSED ");
+            return true;
+        }
+    }
+
+    function checkCertificatePresentInHeader(http:Request req, string apiName, string apiVersion) returns boolean | http:AuthenticationError {
+        printDebug(KEY_AUTHN_FILTER, "Checking the certificate present in the request header.");
```

Comment:
Is the the key (`KEY_AUTHN_FILTER`) we want to use? not a different key for mutualssl handler?

Reply:
All the authentication handlers use the same key